### PR TITLE
feat: add WCPay sections in settings pages

### PR DESCRIPTION
**Pull request by frosso**
Part of https://github.com/Automattic/woocommerce-payments/issues/1576

#### Changes proposed in this Pull Request

Adds the "WooCommerce Payments" and the "All payment methods" sections to the "Payments" tab in WC settings
https://d.pr/i/MrWNFZ

Also ensures that when loading the "Payments" tab, the WCPay settings are displayed by default: https://d.pr/i/bxO9gL
(the page is currently blank because of https://github.com/Automattic/woocommerce-payments/pull/1551 )

#### Testing instructions

- Enable WCPay grouped settings in the dev tools
- Enable Sepa by adding a `_wcpay_feature_sepa` in `wp_options` with value `'1'` (or don't, it doesn't really matter)
- Accessing http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout loads the WCPay settings (the page is currently blank because of https://github.com/Automattic/woocommerce-payments/pull/1551 )
- Accessing http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=all-gateways loads the WC Core Payment settings page
- Saving the settings in http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=all-gateways works and shows the correct page
- By disabling the WCPay grouped settings feature in dev tools, everything goes back to normal

-------------------

- [x] Added changelog entry (or does not apply)
Does not apply - should be behind the scenes
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by frosso** [comment]
Waiting for https://github.com/Automattic/woocommerce-payments/issues/1583 to be merged

**Review by waclawjacek — COMMENTED** [review]
Great work and a really clever solution! 💯

I checked if it might be possible to achieve this with a bit less rewriting and I came up with a suggestion. Let me know what you think!

File: includes/admin/class-wc-payments-admin-sections-overwrite.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	 *
+	 * @param array $sections the sections for the payment gateways tab.
+	 *
+	 * @return array
+	 */
+	public function add_checkout_sections( array $sections ): array {
+		unset( $sections[''] );
+
+		$sections['woocommerce_payments'] = __( 'WooCommerce Payments', 'woocommerce-payments' );
+		$sections['all-gateways']         = __( 'All payment methods', 'woocommerce-payments' );
```
[the remark this answers is not in the document]
Comment (frosso) [reply includes/admin/class-wc-payments-admin-sections-overwrite.php]: Resolved by da995d6

File: includes/admin/class-wc-payments-admin-sections-overwrite.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+		// ensures that on POST (save on the "all gateways" page), the page rendered on redirect is accurate.
+		// this overwrites the hidden "section" input field that is already present on the page.
+		?>
+		<input type="hidden" name="section" value=""/>
+		<?php
+
+		$current_section = '';
+	}
+}
```
[the remark this answers is not in the document]
Comment (frosso) [reply includes/admin/class-wc-payments-admin-sections-overwrite.php]: Thanks for the suggestion! On one hand I like the fact that in one solution the `section` has a name, but in your solution I think things a quite a bit simplified, which I think is a great tradeoff! Done in da995d6

File: includes/admin/class-wc-payments-admin-sections-overwrite.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		add_action( 'woocommerce_settings_page_init', [ $this, 'page_init' ] );
+		add_filter( 'woocommerce_get_sections_checkout', [ $this, 'add_checkout_sections' ] );
+	}
+
+	/**
+	 * Checks the current page in wp-admin to ensure we're looking at the payment gateways settings page.
+	 *
+	 * @return bool
+	 */
+	private function is_current_page_payment_methods() {
```
Comment (waclawjacek) [inline includes/admin/class-wc-payments-admin-sections-overwrite.php:27]: This is not exactly what this method is checking - the check is now more of "was no specific section specified in the URL?". If a section is specified (whether _WooCommerce Payments_ or _All payment methods_), the method will return `false`.

**Review by waclawjacek — APPROVED** [review]
LGTM! :shipit: I suggested one change and found a bug (see below) but I'd say both are non-blocking.

## The bug

I noticed an issue with the up arrow link on a specific payment methods page:

Markup on 2021-04-21 at 15:49:55

After clicking the link, you end up on the WooCommerce Payments section rather than All payment methods.

Page example: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=cod

This is present in both the most recent commit as well as the one with the `$current_section` approach.

It might require changes to individual payment methods.

Since the new section only applies with the feature flag on, I'd say creating a separate issue for this should be enough.

File: includes/admin/class-wc-payments-admin-sections-overwrite.php (hunk 3 above)
Comment (frosso) [reply includes/admin/class-wc-payments-admin-sections-overwrite.php]: Ah, I see what you mean, thanks for pointing it out! Changed 👍

**Comment by frosso** [comment]
@waclawjacek thanks, that's also something I noticed when investigating the spike. Good idea on the issue, here it is: https://github.com/Automattic/woocommerce-payments/issues/1646

I think I have an idea on how to hijack the link, but I don't think we can alter how the breadcrumbs are displayed, due to the hardcoding in here: https://github.com/woocommerce/woocommerce/blob/trunk/includes/abstracts/abstract-wc-payment-gateway.php#L190-L192

**Pull request merged**